core/trino-parser/src/main/java/io/trino/sql/tree/Property.java

```diff
@@ -20,25 +20,46 @@
 import java.util.Optional;
 
 import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkState;
 import static java.util.Objects.requireNonNull;
 
+// TODO: Model DEFAULT as a type of expression to make it easier to do analysis, planning and error reporting in a uniform way
 public class Property
         extends Node
 {
     private final Identifier name;
-    private final Expression value;
+    private final Optional<Expression> value;   // empty iff the value is set to DEFAULT
+
+    /**
+     * Constructs an instance representing a property whose value is set to DEFAULT
+     */
+    public Property(Identifier name)
+    {
+        this(Optional.empty(), name, Optional.empty());
+    }
+
+    /**
+     * Constructs an instance representing a property whose value is set to DEFAULT
+     */
+    public Property(NodeLocation location, Identifier name)
+    {
+        this(Optional.of(requireNonNull(location, "location is null")), name, Optional.empty());
```

Comment:
undo `requireNotNull`. It's sufficient that final constructor checks it

Reply:
The `requireNotNull` in this line checks whether the parameter `NodeLocation location` is `null` (otherwise the call to `Optional.of` may throw a `NullPointerException` without our usual error message before the final constructor is reached). The final constructor checks something different - it checks (via the constructor in the super class) whether the `Optional<NodeLocation>` is null.